### ISSUE TITLE
PEP 3105 Make print a function in twisted/news

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8401]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8401|
|Type|enhancement|
|Created|2016-06-03 10:48:03Z|

This ticket was split out of [ticket:5812] which was too big to review.

PEP 3105 ( ​​​https://www.python.org/dev/peps/pep-3105/ ) **print** deprecates has changed from a statement to a function.

In Python 2.6 and Python 2.7, the default is print as a statement

```
print "Hello"
```

In Python 3.x, the above example is an error, since only print as a function is available.

To consistently use **print** as a function
on Python 2.6, Python 2.7, and Python 3.x, this should be done:

```
from __future__ import print_function

print("Hello")
```

<details><summary>Searchable metadata</summary>

```
trac-id__8401 8401
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__news news
keywords__None None
time__1464950883867695 1464950883867695
changetime__1465739158138024 1465739158138024
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/117

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Thanks for the patch.

Since we are touching this code, I think that is a good time to check whether we really need those print functions in the first place.

I think that they should be converted into logs and not into the print function.

Please consider converting them to logs and resubmit.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

I think that although they should be logs, this code is so unmaintained that it's unlikely this'll happen in a proper way. Twisted's logger also captures stdout, so it's not like this information will be lost.

In the interest of pushing forward so that our codebase can be linted by Python 3-using tools, I'm going to merge this.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset 20315126fde436cda8f8fa4c3d4bb42117cc0b99
```
#!CommitTicketReference repository="" revision="20315126fde436cda8f8fa4c3d4bb42117cc0b99"
Merge 8401-rodrigc-print: Convert from print statement to print() function in twisted/news

Author: rodrigc
Reviewer: hawkowl
Fixes: #8401
```
